router/core/router.go

```diff
@@ -1089,10 +1089,24 @@ func (r *Router) newServer(ctx context.Context, routerConfig *nodev1.RouterConfi
 
 	if r.Config.rateLimit != nil && r.Config.rateLimit.Enabled {
 		handlerOpts.RateLimitConfig = r.Config.rateLimit
-		client := redis.NewClient(&redis.Options{
-			Addr:     r.Config.rateLimit.Storage.Addr,
-			Password: r.Config.rateLimit.Storage.Password,
-		})
+
+		var client *redis.Client
+
+		if r.Config.rateLimit.Storage.Url != "" {
+			options, err := redis.ParseURL(r.Config.rateLimit.Storage.Url)
+
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse the redis connection string %s: %w", r.Config.rateLimit.Storage.Url, err)
+			}
+
+			client = redis.NewClient(options)
+		} else {
```

Comment:
Will this mean that a combination of URL and password is not possible? Or is a password part of the URL?

Reply:
@jensneuse Password can be a part of the url, e.g. `redis://<user>:<password>@<host>:<port>`